File: others/valid_palindrome.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

bool checkPalindrome(string strIn) {
    string strInNew = "";
    for (string::iterator it=strIn.begin(); it != strIn.end(); ++it) {
        if ((*it >= 'a' && *it <= 'z'))
            strInNew += *it;
        else if (*it >= 'A' && *it <= 'Z')
            strInNew += *it-'A' + 'a';
    }
    int len = strInNew.length();
    for (int i = 0; i < len/2; i++) {
        if (strInNew[i] != strInNew[len-i-1]) {
            return false;
        }
        cout<<strInNew[i]<<" "<<strInNew[len-i-1]<<endl;
    }
    return true;
}


int main() {

    string input = "A man, a plan, a canal: Panama";
    cout<<checkPalindrome(input);

    return 0;
}

File: others/valid_palindrome.py
class Solution:
    # @param s, a string
    # @return a boolean
    def isPalindrome(self, s):
        #remove all non-alphanumic chas
        n = len(s)
        l = 0
        r = n - 1
        while l <= r - 1:
            while not s[l].isalpha() and not s[l].isdigit() and l <= r - 1:
                l += 1
            while not s[r].isalpha() and not s[r].isdigit() and l <= r - 1:
                r -= 1

            if s[l].upper() != s[r].upper():
                return False
            l += 1
            r -= 1

        return True

sol = Solution()
print sol.isPalindrome("1a2")
print sol.isPalindrome("A man, a plan, a canal: Panama")
